From 37cf4d78df2f3805f1f230252e1c0030a2f08877 Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <ignatenkobrain@fedoraproject.org>
Date: Sat, 8 Jun 2019 19:35:38 +0200
Subject: [PATCH] Implement %__cargo_skip_build

We need to have an easy way how to skip doing 'cargo build' to speedup a
module builds.

Signed-off-by: Igor Gnatenko <ignatenkobrain@fedoraproject.org>
---
 data/macros.rust-srpm | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/data/macros.rust-srpm b/data/macros.rust-srpm
index 872b87f..88d5a0f 100644
--- a/data/macros.rust-srpm
+++ b/data/macros.rust-srpm
@@ -37,3 +37,19 @@
 \
     print(url .. crate .. '/' .. version .. '/download#/' .. crate .. '-' .. version .. '.crate')
 }
+
+# If crate not in _build_crates and _module_build is set, we should skip the build
+%__cargo_skip_build %{lua:
+local crate = rpm.expand('%{crate}')
+local build_crate = false
+for w in rpm.expand('%{?_build_crates}'):gmatch('%S+') do
+  if w == crate then
+    build_crate = true
+    break
+  end
+end
+if (rpm.expand('%{defined _module_build}') ~= '0' and not build_crate) then
+  print(1)
+else
+  print(0)
+end}
-- 
2.22.0.rc3

